Update Software Catalog setup landing page and add correlation requirements (#31967)
DataDog/documentation: content/en/internal_developer_portal/software_catalog/set_up/_index.md

{{< /whatsnext >}}

By default, Software Catalog is automatically populated with entries discovered from APM, USM, and RUM. You can also manually import entries from other Datadog telemetries, like logs. 

### APM

When you instrument your application code with Datadog APM SDKs or OpenTelemetry, your applications emit traces and generate unsampled trace metrics. These traces and metrics power the entity discovery and dependency mapping capabilities in IDP. Your instrumentation choices (for example, your Datadog Agent version, your SDK version, and whether you use custom instrumentation or service overrides) affect the quality and accuracy of your dependency maps. See [Discover from APM, USM, and RUM][5] for details.

### USM

USM detects Golden Signal metrics (for example, requests, errors, and durations) and maps out application dependencies based on eBPF. It does not require instrumentation of your application code.

### RUM 

RUM provides frontend user experience data, including page performance, errors, session events, and views. If you have RUM applications, they appear in Software Catalog as **Frontend Apps** in the component selector. 

### Other Datadog telemetries

> You can also import entities that are reflected in Datadog telemetries like logs, host metrics, container metrics, network metrics, and process metrics. 

Reviewer: Just a thought to be slightly more active.
```suggestion
You can also import entities that are identified from Datadog telemetries like logs, host metrics, container metrics, network metrics, and process metrics. 
```

When you use [**Import Entities**][10] and choose a data source, Datadog queries that source and searches for valid `DD_SERVICE` tags. Entities are marked with the `kind:service` attribute.

**Note**: You should only do this if the `DD_SERVICE` tags are well maintained and do not contain irrelevant or incorrect tag values.

## Create entities

{{< whatsnext desc=" " >}}
    {{< nextlink href="/internal_developer_portal/software_catalog/set_up/create_entities#through-the-datadog-ui" >}}Create through the Datadog UI{{< /nextlink >}}
    {{< nextlink href="/internal_developer_portal/software_catalog/set_up/create_entities#through-automation" >}}Create through code automation{{< /nextlink >}}
{{< /whatsnext >}}

[Entity definitions][1], defined in entity YAML files, are the canonical source of truth in Software Catalog. You can: 
- Create entity definitions manually through Datadog.
- Store definitions in a version control system like Git, and set up [Source Code Integration][6] to sync definitions with IDP. Changes made to your files are reflected in Datadog within minutes.

> **Note**: Automatic telemetry overlay requires that the name field value matches the corresponding Datadog telemetries keys exactly. See examples for [`kind:datastore`][7], [`kind:queue`][8], and other [entity types][9]. 

Reviewer: Just an edit idea since I think `Datadog telemetries keys` could be a bit unclear for readers.
```suggestion
**Note**: To automatically correlate an entity to its telemetry, the `name` field in your definition must exactly match the primary identifier from the telemetry data. For most services, this is the `service` tag as defined in Datadog's Unified Service Tagging. See examples for [`kind:datastore`][7], [`kind:queue`][8], and other [`entity types`][9].
```

Author: @brett0000FF So much clearer! Thank you!
